server/routes/search.ts

```diff
@@ -65,24 +68,75 @@ export default function routes(router: Router, prisonerSearchService: PrisonerSe
     })
   })
 
-  get('/booking', (req, res) => {
-    const search = req?.body?.search
+  router.get('/visit', (req, res) => {
+    const searchBlock1 = req?.body?.searchBlock1
+    const searchBlock2 = req?.body?.searchBlock2
+    const searchBlock3 = req?.body?.searchBlock3
+    const searchBlock4 = req?.body?.searchBlock4
 
-    res.render('pages/search/booking', { search })
+    res.render('pages/search/visit', { searchBlock1, searchBlock2, searchBlock3, searchBlock4 })
   })
 
-  post('/booking', (req, res) => {
-    const { search } = req.body
+  router.post('/visit', (req, res) => {
+    const { searchBlock1, searchBlock2, searchBlock3, searchBlock4 } = req.body
 
     return res.redirect(
       url.format({
-        pathname: '/search/booking/results',
+        pathname: '/search/visit/results',
         query: {
-          ...(search && { search }),
+          ...(searchBlock1 && { searchBlock1, searchBlock2, searchBlock3, searchBlock4 }),
         },
       })
     )
   })
 
+  router.get('/visit/results', async (req, res) => {
+    const { searchBlock1, searchBlock2, searchBlock3, searchBlock4 } = req.query
+    const search = `${searchBlock1}-${searchBlock2}-${searchBlock3}-${searchBlock4}`
+
+    const currentPage = '1'
```

Comment:
Much is hardwired because it's not a search, it will only ever return a single result.